tests/unit_tests/dft/include/compute_inplace.hpp

```diff
@@ -0,0 +1,160 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_COMPUTE_INPLACE_HPP
+#define ONEMKL_COMPUTE_INPLACE_HPP
+
+#include "compute_tester.hpp"
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+int DFT_Test<precision, domain>::test_in_place_buffer() {
+    if (!init(MemoryAccessModel::buffer)) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::INPLACE);
+
+    const size_t container_size =
+        domain == oneapi::mkl::dft::domain::REAL ? conjugate_even_size : size;
+
+    std::vector<FwdInputType> inout_host(input);
+    sycl::buffer<FwdInputType, 1> inout_buf{ inout_host.data(), sycl::range<1>(container_size) };
```

Comment:
Isn't the size of `inout_host` less than `container_size` for real transforms? (`size` vs `2 * (size / 2 + 1)`) 
This triggers reads past bounds when initializing the buffer data and writes past bound into `inout_host` upon destruction of `inout_buf`

Reply:
Thank you for spotting that. I've corrected `inout_host` to be equal to `container_size` in [6a6d3a7](https://github.com/oneapi-src/oneMKL/pull/261/commits/6a6d3a78b9ecaeea93436f0245a53e1ecf837929).

Reply:
Well, doesn't that trigger the throwing of the newly added `invalid_argument` exception in `reference_forward_dft`, then?

Reply:
It doesn't because the `reference_forward_dft` uses `input` and `out_host_ref` as parameters, instead of `inout_host`. The contents of `input` are copied into `inout_host`.